File: Same-Tree/Same-Tree.py

‘’’
 Definition for a  binary tree node
 class TreeNode:
     def __init__(self, x):
         self.val = x
         self.left = None
         self.right = None
‘’’
class Solution:
    # @param p, a tree node
    # @param q, a tree node
    # @return a boolean
    def isSameTree(self, p, q):
        if p==None and q==None:
            return True
        elif p==None and q!=None or p!=None and q==None:
            return False
        else:
            return self.isSameTree(p.left,q.left) and self.isSameTree(p.right,q.right) and p.val==q.val
        
 

File: Same-Tree/Same-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
    if (null === p && null === q) {
        return true;
    } else if (null !== p && null !== q) {
        return isSameTree(p.left, q.left) && isSameTree(p.right, q.right) && p.val === q.val;
    } else {
        return false;
    }
};